core/server/src/main/java/alluxio/web/WebInterfaceWorkerMetricsServlet.java

```diff
@@ -108,11 +108,39 @@ public boolean matches(String name, Metric metric) {
     });
 
     Map<String, Metric> operations = new TreeMap<>();
-    operations.putAll(counters);
+    for (String key : counters.keySet()) {
+      operations.put(removeHostNameFromWorkerMetricsName(key), counters.get(key));
+    }
     String blockCachedProperty =
         CommonUtils.argsToString(".", workerSource.getName(), WorkerSource.BLOCKS_CACHED);
-    operations.put(blockCachedProperty, mr.getGauges().get(blockCachedProperty));
+    operations.put(removeHostNameFromWorkerMetricsName(blockCachedProperty),
+        mr.getGauges().get(blockCachedProperty));
+
+    Map<String, Counter> rpcInvocationsUpdated = new TreeMap<>();
+    for (String key : rpcInvocations.keySet()) {
+      rpcInvocationsUpdated.put(removeHostNameFromWorkerMetricsName(key), rpcInvocations.get(key));
+    }
+    populateCounterValues(operations, rpcInvocationsUpdated, request);
+  }
 
-    populateCounterValues(operations, rpcInvocations, request);
+  /**
+   * Remove the worker name from worker metrics name to make it easier to retrieve worker metrics
+   * in the worker metrics dashboard.
+   *
+   * @param metricsName the metrics name with hostname builtin
+   * @return the metrics with hostname removed
+   * @throws IOException if the metrics name is invalid
+   */
+  private String removeHostNameFromWorkerMetricsName(String metricsName) throws IOException {
+    String[] pieces = metricsName.split("\\.");
+    if (pieces.length < 3) {
+      throw new IOException("Incorrect worker metrics name: " + metricsName);
```

Comment:
`IllegalArgumentException` is a runtime exception, so this shouldn't be a problem